arbos/addressSet/addressSet.go

```diff
@@ -84,6 +86,44 @@ func (aset *AddressSet) AllMembers(maxNumToReturn uint64) ([]common.Address, err
 	return ret, nil
 }
 
+func (aset *AddressSet) ClearList() error {
+	size, err := aset.size.Get()
+	if err != nil || size == 0 {
+		return err
+	}
+	for i := uint64(1); i <= size; i++ {
+		err = aset.backingStorage.ClearByUint64(i)
+		if err != nil {
+			return err
+		}
+	}
+	return aset.size.Clear()
+}
+
+func (aset *AddressSet) RectifyMapping(addr common.Address) error {
+	isOwner, err := aset.IsMember(addr)
+	if !isOwner || err != nil {
+		return errors.New("RectifyMapping: Address is not an owner")
+	}
+
+	// If the mapping is correct, RectifyMapping shouldnt do anything
+	// Additional safety check to avoid corrupting the mapping after the initial fix
+	addrAsHash := common.BytesToHash(addr.Bytes())
+	slot, _ := aset.byAddress.GetUint64(addrAsHash)
+	atSlot, _ := aset.backingStorage.GetByUint64(slot)
+	if atSlot == addrAsHash && atSlot != common.BytesToHash(common.Address{}.Bytes()) {
+		return nil
```

Comment:
I don't have a super strong opinion on this, but I'm currently thinking that we should return an error here because the address is already correct. This will make sure that this method can't be successfully called indefinitely, which I think makes things simpler.

Reply:
agreed